sourceme.sh

```diff
@@ -1,10 +1,17 @@
 #!/bin/sh
 # Convenience script for setting variables for in-tree use of hammer.
+echo "Installing dependency"
+python3 -m pip install --user typing six enum34 python-jsonschema-objects pyaml pytest
```

Comment:
Could you move this to a separate script maybe named `setup.sh` instead? This script is sourced pretty frequently and it would be good to avoid a potentially expensive run of pip each time.

Reply:
Also, could you pin these to version ranges e.g. like in https://stackoverflow.com/questions/8795617/how-to-pip-install-a-package-with-min-and-max-version-range

Reply:
OK I will use `requirements.txt` to handle the 3rd party package list, and this will introduce 2 more files. :)